### PR TITLE
Visitor navigation #76

### LINKED ISSUE #76: User Story 2, Visitor Navigation
```
As a visitor
I see a navigation bar
This navigation bar includes links for the following:
- a link to return to the welcome / home page of the application ("/")
- a link to browse all items for sale ("/items")
- a link to see all merchants ("/merchants")
- a link to my shopping cart ("/cart")
- a link to log in ("/login")
- a link to the user registration page ("/register")

Next to the shopping cart link I see a count of the items in my cart
```

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #77

child of #79

### PR DESCRIPTION
**Pull request by allieIsOkay**
Adds tests for navigation
Adds visitor navbar partial
Adds logic for rendering partials based on user role to application layout
Adds bootstrap css and JS links to application layout
Adds routes for visitor navigation of site
Adds controllers for visitor routing
Adds empty views for visitor routing
Removes boolean column tests for models
closes #76

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,13 @@
...
+
+  resources :welcome, only: :index
+  resources :items, only: :index
+  resources :users, only: :index
+
+  get '/cart', to: 'cart#show'
+  get '/login', to: 'sessions#new'
+  post '/login', to: 'sessions#create'
+  get '/register', to: 'users#new'
+  post '/reegister', to: 'users#create'
```
Comment (teresa-m-knowles) [inline config/routes.rb:12]: small typo here! should be '/register'

**Review by teresa-m-knowles — APPROVED** [review]
